Formula/dosbox-staging.rb

```diff
@@ -0,0 +1,36 @@
+class DosboxStaging < Formula
+  desc "Modernized DOSBox soft-fork"
+  homepage "https://dosbox-staging.github.io/"
+  url "https://github.com/dosbox-staging/dosbox-staging/archive/v0.75.0.tar.gz"
+  sha256 "6279690a05b9cc134484b8c7d11e9c1cb53b50bdb9bf32bdf683bd66770b6658"
+
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "pkg-config" => :build
+  depends_on "libpng"
+  depends_on :macos => :catalina # Github CI and Apple SDKs limitations
+  depends_on "opusfile"
+  depends_on "sdl2"
+  depends_on "sdl2_net"
+  uses_from_macos "zlib"
+
+  def install
+    args = %W[
+      --prefix=#{prefix}
+      --program-suffix=-staging
+      --disable-debug
+      --disable-silent-rules
+      --disable-dependency-tracking
+      --disable-sdltest
+    ]
+
+    system "./autogen.sh"
+    system "./configure", *args
+    system "make", "install"
+  end
+
+  test do
+    assert_match /dosbox \(dosbox-staging\), version #{version}/,
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
I basically used the testing template an other dosbox fork (dosbox-x, a fork with a completey different target) was using. I thought that that would be fine. Btw. the original dosbox (does not even do any version check). 
Are you sure that are more complex test is necessary here? Dosbox basically starts a gui (a new window with are dos promt) and thats it without any content provided by the user.